docs/page-objects/additional-requests.rst

```diff
@@ -812,26 +126,30 @@ times before giving up:
 
     import attrs
     from tenacity import retry, stop_after_attempt
-    from web_poet import HttpClient, WebPage, validates_input
+    from web_poet import HttpClient, HttpError, WebPage, field
+    from zyte_common_items import Product
+
 
     @attrs.define
-    class MyPage(WebPage):
+    class ProductPage(WebPage[Product]):
         http: HttpClient
 
+        @field
+        def productId(self):
+            return self.css("::attr(product-id)").get()
+
         @retry(stop=stop_after_attempt(3))
-        async def get_data(self):
-            response = await self.http.get("https://toscrape.com/")
-            if not response.css(".expected"):
-                raise ValueError
-            return response.css(".data").get()
-
-        @validates_input
-        async def to_item(self) -> dict:
+        async def get_images(self):
+            return self.http.get(f"https://api.example.com/v2/images?id={self.productId}")
+
+        @field
+        async def images(self):
             try:
-                data = await self.get_data()
-            except ValueError:
-                return {}
-            return {"data": data}
+                response = await self.get_images()
+            except HttpError:
+                return []
+            else:
+                return response.css(".product-images img::attr(src)").getall()
```

Comment:
It doesn't fit the Product schema.
It seems we should add a processor for Product.images.